nbgrader/utils.py

```diff
@@ -226,6 +227,17 @@ def full_split(path):
         return full_split(rest) + (last,)
 
 
+@contextlib.contextmanager
+def chdir(dirname):
+    currdir = os.getcwd()
+    if dirname:
```

Comment:
Please can you remove this ``if`` statement. Let ``chdir`` handle `dirname` not being set properly.

Reply:
If we remove the if statement, we will need to do this later. Is that what you prefer?

```
        if dirname:
            with utils.chdir(dirname):
                nb = self._preprocess(basename)
        else:
            nb = self._preprocess(basename)
```

Reply:
Ah I see now why this is needed, never mind, stick to how it is now :)